app/views/profile/show.html.haml

```diff
@@ -2,7 +2,7 @@
   = "Welcome to #{@user.username}'s garden!"
 .bg-white
   .grid.grid-cols-1.gap-y-10.sm:grid-cols-2.md:grid-cols-4.lg:grid-cols-5.gap-x-6.mt-10
-    - @user.plants.all.each do |plant|
+    - @user.plants.with_attached_img.each do |plant|
```

Comment:
Same here, we can move some part to the controller action and just pass a @user_plant instance variable to the view. makes for a cleaner code.